scenario-compiler/src/main/java/pl/touk/nussknacker/engine/spel/internal/FlinkRowToMapConverter.scala

```diff
@@ -0,0 +1,48 @@
+package pl.touk.nussknacker.engine.spel.internal
+
+import org.springframework.core.convert.{ConversionService, TypeDescriptor}
+import org.springframework.core.convert.converter.{ConditionalGenericConverter, GenericConverter}
+import org.springframework.core.convert.converter.GenericConverter.ConvertiblePair
+import org.springframework.util.ClassUtils
+import pl.touk.nussknacker.engine.util.AssignabilityUtil
+
+import java.util
+import java.util.Collections
+import scala.jdk.CollectionConverters._
+
+// Converts org.apache.flink.types.Row to Map. We should not have library containing org.apache.flink.types.Row class as dependency here
+class FlinkRowToMapConverter(val conversionService: ConversionService) extends ConditionalGenericConverter {
+
+  override def getConvertibleTypes: util.Set[GenericConverter.ConvertiblePair] =
+    Collections.singleton(new ConvertiblePair(classOf[Object], classOf[util.Map[_, _]]))
+
+  override def matches(sourceType: TypeDescriptor, targetType: TypeDescriptor): Boolean =
+    AssignabilityUtil.isAssignableToLoadableClass(
+      sourceType.getObjectType,
+      "org.apache.flink.types.Row"
+    ) && ClassUtils.isAssignable(targetType.getObjectType, classOf[util.Map[_, _]])
+
+  override def convert(source: Any, sourceType: TypeDescriptor, targetType: TypeDescriptor): AnyRef = {
+    if (source == null) {
+      null
+    } else {
+      // if we could import Row class this would look like this:
+//            val r      = source.asInstanceOf[org.apache.flink.types.Row]
+//            val fields = r.getFieldNames(true).asScala
+//            fields.map(e => e -> r.getField(e)).toMap.asJava
+      val getFieldsMethod = source.getClass.getMethod("getFieldNames", classOf[Boolean])
+      getFieldsMethod.setAccessible(true)
+      val fields =
+        getFieldsMethod.invoke(source, (true).asInstanceOf[AnyRef]).asInstanceOf[java.util.Set[String]].asScala
+      fields
```

Comment:
Let's do `.toList` before `.map`. AFAIK `.map` rewrite the result to the same structure as it is invoked on, so in this situation it will do:
```
set = new HashSet
set.add(entry1)
set.add(entry2)
```
Adding to a set involves computation for each entry, including hash computation of potentially deep structures in values. We don't need it, because we don't want to have unique set of entries, we only want to have unique set of keys and values for them.